core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/response/RotateSaltPasswordCheckResponse.java

```diff
@@ -0,0 +1,41 @@
+package com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response;
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.sequenceiq.cloudbreak.api.model.RotateSaltPasswordReason;
+
+import io.swagger.annotations.ApiModel;
+
+@ApiModel
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RotateSaltPasswordCheckResponse {
+
+    private boolean needed;
+
+    private RotateSaltPasswordReason reason;
```

Comment:
Is `needed` really needed? It would be possible to check if rotation is needed by checking if `reason` is null

Reply:
Or this can be a status, and if everything is OK, then a new enum value should be returned like `ACTIVE`

Reply:
That is a great idea! I will rewrite that part